### PR TITLE
feat: experimental use of 'discriminator' for vpng-gw module - `avm/res/network/virtual-network-gateway`

### LINKED ISSUE #1869: [AVM Module Issue]: Virtual Network Gateway no Azure APIPA BGP Setting
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/network/virtual-network-gateway

### (Optional) Module Version

0.1.3

### Description

There is no functionality to configure Azure APIPA address in the module currently

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by fabmas**
## Description

Closes https://github.com/Azure/bicep-registry-modules/issues/1869

Reviewed the code with the use of 'discriminator' for the UDT.
Added the following test-cases:
1. BGP: VPN Active Passive with BGP config but no APIPA
2. BGPAA: VPN Active Active with BGP config but no APIPA
3. BGPAPIPA: VPN Active Passive with BGP config and APIPA 
4. BGPAAAPIPA: VPN Active Active with BGP config and APIPA 
Plus other test-cases without BGP

## Pipeline Reference

[avm.res.network.virtual-network-gateway](https://github.com/fabmas/bicep-registry-modules/actions/workflows/avm.res.network.virtual-network-gateway.yml)

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by AlexanderSehr** [comment]
Hey @fabmas,
I hope you're well! Is still this PR still being worked on or should we close it?

**Comment by eriqua** [comment]
> Hey @fabmas, I hope you're well! Is still this PR still being worked on or should we close it?

Same as the older PR @AlexanderSehr. This is on hold but still active. Moving this to draft as well, but we're not closing them for the moment

**Comment by fabmas** [comment]
Hi @eriqua,

Yes, both are waiting for input. PR #2667 is an experimental version that partially incorporates your suggestion of using UDT with a 'discriminator'. However, I'm not entirely convinced by it, so it would be great to review it together.

The older PR is the initial one that addressed the issue but doesn't include the implementation of UDT with a 'discriminator'. I hope we can review both together soon to help unblock the related issues tied to the same module.

File: avm/res/network/virtual-network-gateway/main.bicep
Comment (eriqua) [inline avm/res/network/virtual-network-gateway/main.bicep:575]: Please mention what would be the default value if not provided

File: avm/res/network/virtual-network-gateway/main.bicep
Comment (eriqua) [inline avm/res/network/virtual-network-gateway/main.bicep:588]: is -pip or -pip2 that will be appended?

File: avm/res/network/virtual-network-gateway/main.bicep
Comment (eriqua) [inline avm/res/network/virtual-network-gateway/main.bicep:566]: Same comment here, is it -pip or -pip2?

File: avm/res/network/virtual-network-gateway/main.bicep
Comment (eriqua) [inline avm/res/network/virtual-network-gateway/main.bicep:591]: Please also add here info about the default value

**Review by eriqua — APPROVED** [review]
:ok:

**Pull request merged**
